api/rql/internal/predicate/action_test.go

```diff
@@ -0,0 +1,65 @@
+package predicate
+
+import (
+	"testing"
+
+	"github.com/puppetlabs/wash/api/rql"
+	"github.com/puppetlabs/wash/api/rql/ast/asttest"
+	"github.com/puppetlabs/wash/api/rql/internal/predicate/expression"
+	"github.com/puppetlabs/wash/plugin"
+	"github.com/stretchr/testify/suite"
+)
+
+type ActionTestSuite struct {
+	asttest.Suite
+}
+
+func (s *ActionTestSuite) TestMarshal() {
+	s.MTC(Action(plugin.ExecAction()), "exec")
+}
+
+func (s *ActionTestSuite) TestUnmarshal() {
+	a := Action(plugin.Action{})
+	s.UMETC(a, 1, "formatted.*<action>", true)
+	s.UMETC(a, "foo", "formatted.*<action>", true)
+	// UMTC doesn't work because s.Equal doesn't work for the Action
+	// type. My best guess is because the Action type has a function
+	// as its field, and s.Equal doesn't work with functions. Thus, we
+	// do our own assertion here.
+	if s.NoError(a.Unmarshal("exec")) {
+		s.True(EqualAction(a, "exec"))
+	}
+}
+
+func (s *ActionTestSuite) TestEvalAction() {
+	a := Action(plugin.ExecAction())
+	s.EAFTC(a, plugin.ListAction())
+	s.EATTC(a, plugin.ExecAction())
+}
+
+func (s *ActionTestSuite) TestExpression_AtomAndNot() {
+	expr := expression.New("action", func() rql.ASTNode {
+		return Action(plugin.Action{})
+	})
+
+	s.MUM(expr, "exec")
+	s.EAFTC(expr, plugin.ListAction())
```

Comment:
Note that when testing `Not`, this becomes `EATTC`. This symmetry (switching `F` and `T` around) is common in all the other predicate and primary tests.